### DISCUSSION TITLE
Linux version for timsTOF (diaPASEF) data

### DISCUSSION THREAD

**Discussion by cjhjhj** [vdemichev/DiaNN#257] [Q&A]
Hello Vadim,

I have downloaded the linux version of version 1.8 (rpm file) and installed it. However, when I executed it, it was shown as version 1.7.12 and looked like not supporting Bruker's .d directory (i.e., timsTOF data). Is there any way to run DIA-NN for timsTOF diaPASEF data in Linux?

Thank you.

**Comment by vdemichev** [comment]
Hi, which link did you use to download? Can you please try again and show the full terminal output?

Best,
Vadim

**Reply by karthik-vm** [reply]
Hi Vadim,

I am helping Ji-Hoon(cjhjhj) who posted this question with the installation of Dia-NN. I initially tried to install via the source code from this 1.8 tag: https://github.com/vdemichev/DiaNN/releases/tag/1.8

While trying to install via source, I encountered missing libpthread.a, libc.a & libm.a since the compilation was done statically. I overcame this by removing the -static flag in the compilation command and I was able to compile diann-linux. But when I ran diann-lniux I see the version is 1.7.12 instead of 1.8 as shown below:

$ diann-linux
DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)
Compiled on Nov 30 2021 14:56:56
Current date and time: Fri Dec  3 15:30:39 2021
Logical CPU cores: 64

0 files will be processed

Later I tried to install via rpm and when I ran the diann-1.8 executable extracted from the rpm I see the below errors. At this time I have loaded the gcc/7.3.0 module.

$ ./diann-1.8
./diann-1.8: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by ./diann-1.8)
./diann-1.8: /lib64/libm.so.6: version `GLIBC_2.23' not found (required by /research/rgs01/applications/hpcf/authorized_apps/rhel7_apps/diann/install/1.8/libtorch_cpu.so)

I tried to build glibc 2.23 and 2.27 from source but using them leads to segfault and other errors.
Any help to build/run the linux version of Dia-NN for timsTOF diaPASEF data is highly appreciated.

Thanks,
Karthik

**Reply by vdemichev** [reply]
Hi Karthik,

Cannot use the sources, only the binaries attached to the release. Github automatically saves a .zip file with the sources, this function cannot be turned off, but the sources are currently not valid and do not correspond to DIA-NN 1.8.

Best,
Vadim

**Reply by vdemichev** [reply]
Yes, you need a Linux version with a relatively recent glibc. Reason: DIA-NN needs C++17 functionality. It's an unfortunate property of Linux that glibc is kind of integrated in it and you cannot just update it to a new version easily.

**Reply by karthik-vm** [reply]
Hi Vadim,

Thanks for your quick response. I will try to install the recent glibc with C++17 functionality and see if that works. I will keep you posted.

Best,
Karthik

**Reply by vdemichev** [reply]
Also, DIA-NN for Linux is compatible with Docker, just use the Ubuntu (18.04) base image

**Reply by karthik-vm** [reply]
Good to know. I will try that route too. Thanks.
